data_steward/cdr_cleaner/cleaning_rules/clean_survey_conduct_recurring_surveys.py

```diff
@@ -0,0 +1,176 @@
+"""
+Inserts data representing the version of a survey into survey_conduct table for surveys with custom ids. Ex COPE/Minute
+
+Recurring versions of surveys such as COPE had all of their questions and answers mapped to a single Module in the PPI
+vocabulary. The same was the case for the Minute surveys. This will make it seem like there is only one version of each
+of these surveys. Researchers need to know the version of the survey the participant took. This CR will use
+`cope_survey_semantic_version_map` which is provided to curation from the rdr team, to update the survey version in
+survey_source_concept_id and survey_source_value for each observation.
+
+There are multiple cleaning rules that clean the survey conduct table.
+This CR will clean survey_concept_id and survey_source_value for the surveys with custom concept ids.
+Another cleans survey_source_concept_id for all valid records.
+Then observations that do not have a valid survey will be dropped in another CR.
+
+Original Issues: DC-3082
+"""
+# Python imports
+import logging
+
+# Project imports
+from cdr_cleaner.cleaning_rules.base_cleaning_rule import BaseCleaningRule, query_spec_list
+from constants.cdr_cleaner import clean_cdr as cdr_consts
+from common import JINJA_ENV, SURVEY_CONDUCT, COPE_SURVEY_MAP
+
+LOGGER = logging.getLogger(__name__)
+
+JIRA_ISSUE_NUMBERS = ['DC-3082']
+
+DOMAIN_TABLES = [SURVEY_CONDUCT]
+REFERENCE_TABLES = [COPE_SURVEY_MAP]
+
+SANDBOX_SURVEY_CONDUCT = JINJA_ENV.from_string("""
+CREATE OR REPLACE TABLE `{{project_id}}.{{sandbox_dataset_id}}.{{sandbox_table_id}}` AS (
+SELECT * FROM `{{project_id}}.{{dataset_id}}.survey_conduct`
+WHERE survey_conduct_id IN (SELECT questionnaire_response_id FROM `{{project_id}}.{{dataset_id}}.{{cope_survey_map}}`)
+AND (survey_concept_id NOT IN (2100000006, 2100000007, 2100000002, 2100000005, 2100000004, 2100000003, 905047, 905055, 765936, 1741006)
```

Comment:
Will this list be expanded in the future? (e.g., We get cope surveys with new `cope_month` that is not listed here)
If yes, this CR will unintentionally sandbox the new surveys. `CLEAN_SURVEY_CONDUCT` will not update such records (those will fall into `ELSE`), so it will not be a problem after all.

So, it is not a problem, but I wanted to mention it here for clarity.

Reply:
That I am not sure of. I believe we will not get anymore COPE or Minute surveys but it is very possible that we get more recurring surveys in the future. If other recurring surveys do get added and they do map to the same Module code then I don't know if a new mapping table would be created or the one used here would be expanded. 
I think the plan is to map  any new recurring surveys to different module codes so this **shouldn't** be an issue, but I plan to look into the raw_rdr_export qc notebook. If a check does not exist for new surveys or differences in cope_month, etc. then I'll make a ticket and get it added. Listing the CRs that will need an update if it/they fail.